pkg/services/sqlstore/annotation.go

```diff
@@ -268,3 +268,53 @@ func (r *SQLAnnotationRepo) Delete(params *annotations.DeleteParams) error {
 		return nil
 	})
 }
+
+func (r *SQLAnnotationRepo) FindTags(query *annotations.TagsQuery) ([]*annotations.TagsDTO, error) {
+	if query.Limit == 0 {
+		query.Limit = 100
+	}
+
+	var sql bytes.Buffer
+	params := make([]interface{}, 0)
+	tagKey := `tag.` + dialect.Quote("key")
+	tagValue := `tag.` + dialect.Quote("value")
+
+	sql.WriteString(`
+		SELECT
+			` + tagKey + `,
+			` + tagValue + `,
+			count(*) as count
+		FROM tag
+		INNER JOIN annotation_tag ON tag.id = annotation_tag.tag_id
+		INNER JOIN annotation ON annotation_tag.annotation_id = annotation.id
+`)
+
+	sql.WriteString(`WHERE annotation.org_id = ?`)
+	params = append(params, query.OrgID)
+
+	sql.WriteString(` AND (` + tagKey + ` ` + dialect.LikeStr() + ` ? OR ` + tagValue + ` ` + dialect.LikeStr() + ` ?)`)
+	params = append(params, `%`+query.Tag+`%`, `%`+query.Tag+`%`)
+
+	sql.WriteString(` GROUP BY ` + tagKey + `,` + tagValue)
+	sql.WriteString(` ORDER BY ` + tagKey + `,` + tagValue)
+	sql.WriteString(` ` + dialect.Limit(query.Limit))
+
+	var items []*annotations.Tags
+	if err := x.SQL(sql.String(), params...).Find(&items); err != nil {
+		return nil, err
+	}
+
+	tags := make([]*annotations.TagsDTO, 0)
+	for _, item := range items {
+		tag := item.Key
+		if len(item.Value) > 0 {
+			tag = item.Key + ":" + item.Value
```

Comment:
I have some problems understand how your going to use this. If you would allow user to select a tag from a list and separately the value of it then it would make more sense to me to return Tag and Value as separate fields? Or maybe if you want to separately provide auto-completion for tags and values might make sense with two endpoints, one for tag keys and one for tag values by key? I probably didn't even know that you could have both key and value tags for annotations :shrug:

Reply:
Sorry for the bad explanation, we want to create a component where the user can choose existing tags or add new ones. Inspired by this: 
![image](https://user-images.githubusercontent.com/562238/123900414-53767e00-d969-11eb-80ab-946f1ad4b581.png)

@ashharrison90 I just realized that the component would need to have add new tag functionality too (not just listing existing ones) when creating tags manually in the graph. (the Annotation Editor work in TimeSeries)

Reply:
Yes I understand this, but above screenshot only list tag keys where's you're going to list tag key + value as `<key>:<value>`? I guess my question was whether you planned to have a drop down for tag keys and then when key has been selected a drop down for tag values would auto-complete :smile: But I guess you're planning to use `<key>:<value>`.